src/components/Timeline/EventFilter.js

```diff
@@ -1,22 +1,107 @@
+import * as formActions from '../../actions/formActions'
 import * as eventActions from '../../actions/eventActions'
+import * as filterActions from '../../actions/filterActions'
 
-const EventFilter = ({pagination, dispatch}) =>  {
-  return <div className="incident-EventFilter">
-          <IconButtonStyled
-            tooltip='order'
-            onTouchTap={() => dispatch(eventActions.pagination.sort('occurred'))}
-          >
-            {
-              pagination && pagination.order === 'desc'
-              ? <ArrowDown/>
-              : <ArrowUp/>
-            }
-          </IconButtonStyled>
-        </div>
-}
-
-export default EventFilter
\ No newline at end of file
+
+export const dataSourceConfig = {
+  text: 'name',
+  value: 'id'
+}
+
+export const chipStyles = {
+  chip: {
+    margin: 4
+  },
+  wrapper: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  }
+}
+
+export const filterSearchForm = {
+  name: 'filter selection',
+  field: 'input'
+}
+
+
+const EventFilter = ({pagination, filter, filterSearchField, eventTypes, filterTypes, dispatch, history}) =>  {
+  const filterChips = filter && filter.eventTypes && eventTypes ? renderChips(history, filter, eventTypes, dispatch): null
+  return  (
+    <div className="incident-EventFilter">
+      {filterChips}
+      <AutoComplete
+        floatingLabelText="Filter by event type"
+        filter={AutoComplete.caseInsensitiveFilter}
+        dataSource={filterTypes}
+        searchText={filterSearchField || ''}
+        onUpdateInput={(searchText) => dispatch(formActions.updateInput(filterSearchForm.name, filterSearchForm.field, searchText))}
+        onNewRequest={
+          (eventType, indexInDataSource) => {
+            dispatch(filterActions.addFilter(history)(filter, eventType))
+            dispatch(formActions.clearInput(filterSearchForm.name, filterSearchForm.field))
+          }
+        }
+        dataSourceConfig={dataSourceConfig}
+      />
+      <IconButtonStyled
+        tooltip='order'
+        onTouchTap={() => dispatch(eventActions.pagination.sort('occurred'))}
+      >
+        {
+          pagination && pagination.order === 'desc'
+          ? <ArrowDown/>
+          : <ArrowUp/>
+        }
+      </IconButtonStyled>
+    </div>
+  )
+}
+  
+  
+const renderChips = (history, filter, eventTypes, dispatch) => {
+  return (
+    <div style={chipStyles.wrapper}>
+      {filter.eventTypes.map((passedEventType) => renderChip(history, filter, eventTypes, passedEventType, dispatch))}
+    </div>
+  )
+}
+
+const renderChip = (history, filter, eventTypes, passedEventType, dispatch) => {
+  const eventType = filterActions.findEventTypeInRef(passedEventType, eventTypes)
+  return (
+    <Chip
+      key={eventType.id}
+      onRequestDelete={() => dispatch(filterActions.removeFilter(history)(filter, eventType))}
+      style={chipStyles.chip}
+    >
+      {eventType.name}
+    </Chip>
+  )
+}
+
+const extractEventTypesFromEventTypesObject = (eventTypes) => {
```

Comment:
Avoid using "for ... in" (see https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea
) and avoid using "var" (see https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var-to-declare-a-variable).

This method can be rewritten using .map() instead of a loop.

Reply:
It's not an array.  It's going through and pulling items from an object.  I don't think "map" works with that.  Do you know of a better way to iterate through an object?

Reply:
You can map an object using Underscore: http://underscorejs.org/#map and I bet there's some utilities built into React for this as well.

Reply:
extractEventTypesFromEventTypesObject(eventTypes) can be replaced by Object.values(eventTypes)

Reply:
done